gateway/src/http_entrypoints.rs

```diff
@@ -116,6 +117,39 @@ pub enum SpIgnition {
     },
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize, JsonSchema)]
+struct SpUpdateStatus {
+    id: Uuid,
+    state: SpUpdateState,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize, JsonSchema)]
+#[serde(tag = "state", rename_all = "snake_case")]
+enum SpUpdateState {
+    /// The SP is preparing to receive an update.
+    ///
+    /// May or may not include progress, depending on the capabilities of the
+    /// component being updated. If progress is present, the units are defined
+    /// by the SP.
+    Preparing { progress: Option<Progress> },
+    /// The SP is currently receiving an update.
+    ///
+    /// The units of `progress` are bytes.
+    InProgress { progress: Progress },
+    /// The SP has completed receiving an update.
+    Complete,
+    /// The SP has aborted an in-progress update.
+    Aborted,
+}
+
+/// Progress of an operation; units of `current` and `total` are defined by the
+/// operation.
```

Comment:
above we say that the units are defined by the SP or that they're bytes. If the definition really varies by the variant, perhaps it would be clearer to have these fields in the variants themselves..

Reply:
Reworked slightly in 4b09d42f, making it clear that `InProgress` is bytes and preparation progress is in undefined units.